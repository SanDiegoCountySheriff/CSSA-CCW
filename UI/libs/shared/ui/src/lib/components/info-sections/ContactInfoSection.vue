<template>
  <v-container
    fluid
    class="info-section-container rounded"
  >
    <v-banner class="sub-header font-weight-bold text-left my-5 pl-0">
      {{ $t('Contact Information: ') }}
      <template #actions>
        <v-tooltip bottom>
          <template #activator="{ on, attrs }">
            <v-btn
              icon
              @click="handleEditRequest"
              v-bind="attrs"
              v-on="on"
            >
              <v-icon :color="$vuetify.theme.dark ? 'info' : 'info'">
                mdi-square-edit-outline
              </v-icon>
            </v-btn>
          </template>
          {{ $t('Edit Section') }}
        </v-tooltip>
      </template>
    </v-banner>
    <v-row>
      <v-col
        cols="12"
        lg="6"
      >
        <v-banner
          rounded
          single-line
          class="text-left"
        >
          <v-icon
            left
            color="accent"
          >
            mdi-card-account-phone
          </v-icon>
          <strong>
            {{ $t('Primary Phone Number: ') }}
          </strong>
          {{ props.contactInfo.primaryPhoneNumber }}
        </v-banner>
      </v-col>
      <v-col
        cols="12"
        lg="6"
      >
        <v-banner
          rounded
          single-line
          class="text-left"
        >
          <v-icon
            left
            color="accent"
          >
            mdi-card-account-phone
          </v-icon>
          <strong>
            {{ $t('Cell Phone Number: ') }}
          </strong>
          {{ props.contactInfo.cellPhoneNumber }}
        </v-banner>
      </v-col>
    </v-row>

    <v-row>
      <v-col
        cols="12"
        lg="6"
      >
        <v-banner
          rounded
          single-line
          class="text-left"
        >
          <v-icon
            left
            color="accent"
          >
            mdi-card-account-phone
          </v-icon>
          <strong>
            {{ $t('Fax Phone Number: ') }}
          </strong>
          {{ props.contactInfo.faxPhoneNumber }}
        </v-banner>
      </v-col>

      <v-col
        cols="12"
        lg="6"
      >
        <v-banner
          rounded
          single-line
          class="text-left"
        >
          <v-icon
            left
            color="accent"
          >
            mdi-card-account-phone
          </v-icon>
          <strong>
            {{ $t('Work Phone Number: ') }}
          </strong>
          {{ props.contactInfo.workPhoneNumber }}
        </v-banner>
      </v-col>
    </v-row>
  </v-container>
</template>

<script setup lang="ts">
import { ContactInfoType } from '@shared-utils/types/defaultTypes'
import { useCompleteApplicationStore } from '@shared-ui/stores/completeApplication'
import { useRouter } from 'vue-router/composables'

interface IContactInfoSectionProps {
  contactInfo: ContactInfoType
  color: string
}

const props = defineProps<IContactInfoSectionProps>()
const router = useRouter()
const applicationStore = useCompleteApplicationStore()

function handleEditRequest() {
  applicationStore.completeApplication.application.currentStep = 5
  router.push({
    path: '/form',
    query: {
      applicationId: applicationStore.completeApplication.id,
      isComplete: applicationStore.completeApplication.application.isComplete,
    },
  })
}
</script>

<style lang="scss" scoped>
.info-section-container {
  width: 80%;
  height: 100%;
  margin: 0;
  padding: 0;
}
.info-row {
  display: flex;
  flex-direction: row;
  max-height: 2vh;
  min-height: 1vh;
  margin-left: 0.5rem;
}

.info-text {
  margin-left: 0.5rem;
  text-align: start;
  height: 1.8em;
  width: 50%;
  margin-bottom: 0.5rem;
  padding-bottom: 0.5rem;
  padding-left: 1rem;
  background-color: rgba(211, 241, 241, 0.3);
  border-bottom: 1px solid #666;
  border-radius: 5px;
  font-size: 1.2em;
  font-weight: bold;
}
</style>
