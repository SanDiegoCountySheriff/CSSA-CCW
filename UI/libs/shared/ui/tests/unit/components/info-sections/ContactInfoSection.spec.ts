/* eslint-disable @typescript-eslint/ban-ts-comment */
import ContactInfoSection from '@shared-ui/components/info-sections/ContactInfoSection.vue'
import Vuetify from 'vuetify'
import { createLocalVue, mount } from '@vue/test-utils'

const localVue = createLocalVue()

describe('ContactInfoSection', () => {
  let vuetify
  let wrapper

  beforeEach(() => {
    vuetify = new Vuetify()
    //@ts-ignore
    wrapper = mount(ContactInfoSection, {
      localVue,
      vuetify,
      mocks: {
        $t: text => text,
      },
      propsData: {
        contactInfo: {
          primaryPhoneNumber: 'primary',
          cellPhoneNumber: 'cell',
          workPhoneNumber: 'work',
          faxPhoneNumber: 'fax',
          textMessageUpdates: true,
        },
      },
    })
  })

  it('Should match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
