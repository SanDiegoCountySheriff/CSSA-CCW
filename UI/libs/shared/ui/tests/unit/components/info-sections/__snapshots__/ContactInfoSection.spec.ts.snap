// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactInfoSection Should match the snapshot 1`] = `
<div class="container info-section-container rounded container--fluid">
  <div class="rounded v-card v-sheet theme--light">
    <div class="v-card__title ml-3">
      Contact Information
    </div>
    <div class="v-card__text m-5">
      <div class="row">
        <div class="col">
          <div class="info-row">
            <h3>Primary phone number</h3>
            <p class="info-text">
              primary
            </p>
          </div>
        </div>
        <div class="col">
          <div class="info-row">
            <h3>Cell phone number</h3>
            <p class="info-text">
              cell
            </p>
          </div>
        </div>
      </div>
      <div class="row">
        <div class="col">
          <div class="info-row">
            <h3>Work phone number</h3>
            <p class="info-text">
              work
            </p>
          </div>
        </div>
        <div class="col">
          <div class="info-row">
            <h3> Fax number</h3>
            <p class="info-text">
              fax
            </p>
          </div>
        </div>
      </div>
      <div class="row">
        <div class="col">
          <div class="info-row">
            <h3>Text message update</h3>
            <p class="info-text">
              true
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
